jax/_src/lax/lax.py

```diff
@@ -5346,3 +5346,59 @@ def convert_to(other_dtype, bint_dtype) -> bool:
 
 
 core.bint._rules = BIntRules
+
+
+def optimization_barrier(operand, /):
+  """Prevents the compiler from moving operations across the barrier.
+
+  Optimization barriers have a number of possible uses:
+
+  * An optimization barrier ensures that all inputs are evaluated before any
+    operators that depend on the barrier's outputs. This can be used to enforce
```

Comment:
Do we clearly define "operator" somewhere? Whenever I see it I wonder if we should say "computations" or "function applications".